### PR TITLE
Add  e2e compatibility test

### PR DESCRIPTION
**Pull request by wlq1212**
**What type of PR is this?**:

/kind feature

**What this PR does / why we need it**:
Add  k8s compatibility test

**Which issue(s) this PR fixes**:
Fixes #https://github.com/kubeedge/kubeedge/issues/4919

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```
NONE
```

### CODE REVIEW

**Review by fisherxu — COMMENTED** [review]
The PR need be rebased :) And in commit, looks like github can't associate with your account， you need to add the email addr to your email list in settings. @wlq1212

**Comment by wlq1212** [comment]
> The e2e test CI status is expected. You can try config an object to matrix:
> 
> ```yaml
> strategy:
>       matrix:
>         cases:
>           - protocol: ""
>             version: ""
>           - protocol: ""
>             version: ""
> ```



> The e2e test CI status is expected. You can try config an object to matrix:
> 
> ```yaml
> strategy:
>       matrix:
>         cases:
>           - protocol: ""
>             version: ""
>           - protocol: ""
>             version: ""
> ```

Both ways, the result is the same

File: .github/workflows/main.yaml [hunk 1]
```diff
@@ -204,7 +214,8 @@ jobs:
       # for WebSocket protocol, we will use containerd as edgecore container runtime
       # just for covering both docker and CRI runtime e2e cases
       - run: |
-          export PROTOCOL=${{ matrix.PROTOCOL }}
+          export PROTOCOL=${{ matrix.protocol }}
+          export KIND_IMAGE=${{ matrix.protocol }}
```
Comment (Shelley-BaoYue) [inline .github/workflows/main.yaml:218]: matrix.version?

Comment (wlq1212) [reply .github/workflows/main.yaml]: done

**Comment by Shelley-BaoYue** [comment]
Is it better to use  ${{ matrix.cases.version }}-${{ matrix.case.PROTOCOL }}-e2e-test-logs?

image

**Comment by wlq1212** [comment]
Artifact name has restrictions  lead to e2e test error, As shown below:
企业微信截图_96f96c08-0048-4a0e-a139-0ab0ba05926c
@Shelley-BaoYue

**Comment by WillardHu** [comment]
> Is it better to use ${{ matrix.cases.version }}-${{ matrix.case.PROTOCOL }}-e2e-test-logs?
> 
> image

You can set version to the value of the version type, likes:
```yaml
cases:
  - protocol: ""
    version: v1.22.0
  ...
```
And append export value `export KIND_IMAGE=kindest/node:${{ matrix.cases.version }}`  @wlq1212

**Comment by wlq1212** [comment]
> Is it better to use ${{ matrix.cases.version }}-${{ matrix.case.PROTOCOL }}-e2e-test-logs?

have been added

**Comment by fisherxu** [comment]
/lgtm
/approve

Thanks for the great tests! 

/hold
Leave  @WillardHu @Shelley-BaoYue  for final review.

**Comment by fisherxu** [comment]
After PR merged, I'll config the `required` for the actions.

**Comment by WillardHu** [comment]
> /lgtm /approve
> 
> Thanks for the great tests!
> 
> /hold Leave @WillardHu @Shelley-BaoYue for final review.

lgtm

**Pull request merged**